src/git-version.cr

```diff
@@ -10,12 +10,19 @@ module GitVersion
 
   DEV_BRANCH_SUFFIX = "SNAPSHOT"
 
-  MAJOR_BUMP_COMMENT = "breaking:"
-  MINOR_BUMP_COMMENT = "feature:"
-
   class Git
-    def initialize(@dev_branch : String, @release_branch : String = "master",
+    def initialize(@dev_branch : String, @release_branch : String, @minor_identifier : String, @major_identifier : String,
                    @folder = FileUtils.pwd, @prefix : String = "", @log_paths : String = "")
+      @major_id_is_regex = false
+      @minor_id_is_regex = false
+      if match = /\/(.*)\//.match(@major_identifier)
```

Comment:
I am not 100% sure what this code is supposed to do but a regex can be any string and does not need to be between slashes. Seems like we are making this simple tool very complicated to accommodate this use case.
What is the scenario where you need a regex?

Reply:
Basically what it does is check whether the input starts and ends with a slash and then extracts the relevant regex.

As to why regex can be useful: simple string matching can be kinda restrictive in regards to the various identifiers used to identify minor and major releases, forcing contributes to use certain strings in their commit messages.

Reply:
For example if someone wanted to match feat and feature they could use /feat(?:ure)?:/ instead of having to specify both

Reply:
Sounds like a reasonable use case. Can we then add a note to the readme explaining that the regexp has to be delimited with slashes?

Reply:
Done ☑️